lib/linalg/BasisWriter.cpp

```diff
@@ -53,20 +53,19 @@ BasisWriter::BasisWriter(
     char tmp2[100];
     sprintf(tmp2, "_snapshot.%06d", rank);
     snap_file_name = base_file_name + tmp2;
+
+    // create and open snapshot/basis database
+    // TODO(kevin): can it be CSV at all? we might want to remove if statement here.
+    if (db_format_ == Database::HDF5) {
+        d_snap_database = new HDFDatabase();
+        d_database = new HDFDatabase();
+    }
 }
 
 BasisWriter::~BasisWriter()
 {
-    if (d_database) {
-        d_database->putInteger("num_time_intervals", d_num_intervals_written);
-        d_database->close();
-        delete d_database;
-    }
-    if (d_snap_database) {
-        d_snap_database->putInteger("num_time_intervals", d_num_intervals_written);
-        d_snap_database->close();
-        delete d_snap_database;
-    }
+    if (d_database) delete d_database;
```

Comment:
There is no need to check `if (d_database)`, since it is initialized by the constructor to `NULL` (which could be changed to `nullptr`) and can be deleted in any case.